Tests/ApolloCodegenTests/CodeGeneration/Templates/SelectionSet/SelectionSetTemplateTests.swift

```diff
@@ -0,0 +1,459 @@
+import XCTest
+import Nimble
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class SelectionSetTemplateTests: XCTestCase {
+
+  var schemaSDL: String!
+  var document: String!
+  var ir: IR!
+  var operation: IR.Operation!
+  var subject: SelectionSetTemplate!
+
+  override func setUp() {
+    super.setUp()
+  }
+
+  override func tearDown() {
+    schemaSDL = nil
+    document = nil
+    operation = nil
+    subject = nil
+    super.tearDown()
+  }
+
+  // MARK: - Helpers
+
+  func buildSubjectAndOperation(named operationName: String = "TestOperation") throws {
+    ir = try .mock(schema: schemaSDL, document: document)
+    let operationDefinition = try XCTUnwrap(ir.compilationResult[operation: operationName])
+    operation = ir.build(operation: operationDefinition)
+    subject = SelectionSetTemplate(schema: ir.schema)
+  }
+
+  // MARK: - Tests
+
+  func test__render_rendersClosingBracket() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    type Animal {
+      species: String!
+    }
+    """
+
+    document = """
+    query TestOperation {
+      allAnimals {
+        species
+      }
+    }
+    """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let allAnimals = try XCTUnwrap(
+      operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField
+    )
+
+    let actual = subject.render(field: allAnimals)
+
+    // then
+    expect(String(actual.reversed())).to(equalLineByLine("}", ignoringExtraLines: true))
+  }
+
+  // MARK: Parent Type
+
+  func test__render_parentType__givenParentTypeAs_Object_rendersParentType() throws {
```

Comment:
I think the focus of these tests would be clearer if they were named `__render_field__` vs. `__render_parentType__`. `rendersParentType` is clear as to the expected outcome.